feature/gribi/mpls_in_udp/otg_tests/mpls_in_udp_scale/mpls_in_udp_scale_test.go

```diff
@@ -0,0 +1,1757 @@
+			packetValid = false
+		}
+
+		// Validate UDP header - extract raw bytes for robust parsing
+		udpHeaderBytes := udpLayer.LayerContents()
+		t.Logf("Packet %d: UDP header bytes: %X", packetCount, udpHeaderBytes)
+
+		if len(udpHeaderBytes) < 8 {
+			t.Errorf("Packet %d: UDP header too short (len: %d)", packetCount, len(udpHeaderBytes))
+			packetValid = false
+		}
+
+		// Validate MPLS header inside UDP payload
+		payload := udpLayer.LayerPayload()
+		if len(payload) < 4 {
+			t.Errorf("Packet %d: UDP payload too short for MPLS header, len=%d", packetCount, len(payload))
+			packetValid = false
+		} else {
+			mplsHeaderVal := binary.BigEndian.Uint32(payload[:4])
+			label := (mplsHeaderVal >> 12) & 0xFFFFF
+			bottomOfStack := (mplsHeaderVal >> 8) & 0x1
+			mplsTTL := mplsHeaderVal & 0xFF
+			t.Logf("Packet %d: %s", packetCount, formatMPLSHeader(payload[:4]))
+
+			if len(labelList) != 0 {
+				// Validate that label is in labelList
+				found := false
+				for _, l := range labelList {
+					if uint64(label) == l {
+						found = true
+						break
+					}
+				}
+				if !found {
+					t.Errorf("Packet %d: Got MPLS Label %d, want one of %v", packetCount, label, labelList)
+					packetValid = false
+				}
+			} else {
+				// Validate exact match with pr.mplsLabel
+				if uint64(label) != pr.mplsLabel {
+					t.Errorf("Packet %d: Got MPLS Label %d, want %d", packetCount, label, pr.mplsLabel)
+					packetValid = false
+				}
+			}
+			if bottomOfStack != 1 {
+				t.Errorf("Packet %d: Got MPLS Bottom of Stack bit %d, want 1", packetCount, bottomOfStack)
+				packetValid = false
+			}
+			expectedMPLSTTL := ttl - 1 // Inner packet TTL decremented by 1
+			if uint32(mplsTTL) != expectedMPLSTTL {
+				t.Errorf("Packet %d: Got MPLS TTL %d, want %d", packetCount, mplsTTL, expectedMPLSTTL)
+				packetValid = false
+			}
+		}
+
+		if packetValid {
+			validMplsPacketCount++
+			if validMplsPacketCount <= 2 {
+				t.Logf("Packet %d: MPLS validation PASSED", packetCount)
+			}
+		} else {
+			t.Logf("Packet %d: MPLS validation FAILED", packetCount)
+		}
+	}
+
+	// Summary and validation results
+	t.Logf("=== PACKET CAPTURE VALIDATION SUMMARY ===")
+	t.Logf("Total packets captured: %d", packetCount)
+	t.Logf("UDP-IPv6 packets found: %d", mplsPacketCount)
+	t.Logf("Valid MPLS-in-UDP packets: %d", validMplsPacketCount)
+
+	if packetCount == 0 {
+		t.Errorf("No packets captured on port %s", otgPortName)
+	} else if mplsPacketCount == 0 {
+		t.Errorf("No UDP-IPv6 packets found in capture on port %s", otgPortName)
+	} else if validMplsPacketCount == 0 {
+		t.Errorf("No valid MPLS-in-UDP packets found in capture on port %s", otgPortName)
+	} else if validMplsPacketCount < (mplsPacketCount / 2) {
+		t.Errorf("Many packets (%d/%d) failed validation", mplsPacketCount-validMplsPacketCount, mplsPacketCount)
+	} else {
+		t.Logf("Packet capture validation PASSED: Found %d valid MPLS-in-UDP packets", validMplsPacketCount)
+	}
```

Comment:
A `switch` is more idiomatic in Go - https://go.dev/doc/effective_go#switch


```suggestion
	switch {
	case packetCount == 0:
		t.Errorf("No packets captured on port %s", otgPortName)
	case mplsPacketCount == 0:
		t.Errorf("No UDP-IPv6 packets found in capture on port %s", otgPortName)
	case validMplsPacketCount == 0:
		t.Errorf("No valid MPLS-in-UDP packets found in capture on port %s", otgPortName)
	case validMplsPacketCount < (mplsPacketCount / 2):
		t.Errorf("Many packets (%d/%d) failed validation", mplsPacketCount-validMplsPacketCount, mplsPacketCount)
	default:
		t.Logf("Packet capture validation PASSED: Found %d valid MPLS-in-UDP packets", validMplsPacketCount)
	}
```